forge-gui/res/cardsfolder/upcoming/volatile_stormdrake.txt

```diff
@@ -0,0 +1,14 @@
+Name:Volatile Stormdrake
+ManaCost:1 U
+Types:Creature Drake
+PT:3/2
+K:Flying
+K:Hexproof:Activated,Triggered:activated and triggered abilities
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigExchange | TriggerDescription$ When CARDNAME enters the battlefield, exchange control of CARDNAME and target creature an opponent controls. If you do, you get {E}{E}{E}{E}, then sacrifice that creature unless you pay an amount of {E} equal to its mana value.
+SVar:TrigExchange:DB$ ExchangeControl | Defined$ Self | ValidTgts$ Creature.OppCtrl | TgtPrompt$ Select target creature an opponent controls | CantFizzle$ True | RememberExchanged$ 1 | SubAbility$ Energy
+SVar:Energy:DB$ PutCounter | ConditionDefined$ Remembered | ConditionPresent$ Card | Defined$ You | CounterType$ ENERGY | CounterNum$ 4 | SubAbility$ Sacrifice
+SVar:Sacrifice:DB$ Sacrifice | ConditionDefined$ Remembered | ConditionPresent$ Card | Defined$ Remembered | UnlessCost$ PayEnergy<N> | UnlessPayer$ You | SubAbility$ Cleanup
```

Comment:
why not just Defined$ Targeted?
no reason for extra logic in ControlExchange 🤔

Reply:
Ah good point

I changed ControlExchange first
Then realized the issue was really that SacrificeEffect couldn't handle "Defined" well